filters/imagefilter.go

```diff
@@ -48,25 +49,12 @@ func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
 
 	rsp.Header.Del("Content-Length")
 
-	in := rsp.Body
-	r, w := io.Pipe()
-	rsp.Body = r
-
-	go handleImageTransform(w, in, f)
+	handleImageTransform(rsp, f)
 }
 
-func handleImageTransform(out *io.PipeWriter, in io.ReadCloser, f ImageFilter) error {
+func handleImageTransform(rsp *http.Response, f ImageFilter) error {
 	var err error
-
-	defer func() {
```

Comment:
not sure how ioutil.NopCloser works so - are you sure we can delete that? on error the stream (in/out) gets properly closed?